src/DGtal/math/MeaningfulScaleAnalysis.h

```diff
@@ -0,0 +1,299 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file MeaningfulScaleAnalysis.h
+ * @author Bertrand Kerautret (\c kerautre@loria.fr )
+ * LORIA (CNRS, UMR 7503), University of Nancy, France
+ *
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5807), University of Savoie, France
+ * 
+ *
+ * @date 2015/12/27
+ *
+ * Header file for module MeaningfulScaleAnalysis.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(MeaningfulScaleAnalysis_RECURSES)
+#error Recursive header files inclusion detected in MeaningfulScaleAnalysis.h
+#else // defined(MeaningfulScaleAnalysis_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define MeaningfulScaleAnalysis_RECURSES
+
+#if !defined MeaningfulScaleAnalysis_h
+/** Prevents repeated inclusion of headers. */
+#define MeaningfulScaleAnalysis_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/math/Profile.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // class MeaningfulScaleAnalysis
+  /**
+   * Description of class 'MeaningfulScaleAnalysis' <p> \brief Aim:
+   * This class proposes the implementation of different methods used
+   * to define the meaningful scale analysis as proposed in 
+   * \cite kerautret_meaningful_2012 . In particular, it exploits the
+   * Profile class to represent a multi-scale profile and to compute a
+   * meaningful scale. It also permits to get a noise estimation from
+   * the given profile.
+   * 
+   *
+   * A typical use is to exploit from the length of maximal segments
```

Comment:
A typical example is for instance the length of maximal segments obtained at different scales